src/java/com/google/tagpost/TagpostService.java

```diff
@@ -1,21 +1,85 @@
 package com.google.tagpost;
 
-import com.google.tagpost.FetchMessageRequest;
-import com.google.tagpost.FetchMessageResponse;
-import com.google.tagpost.TagpostServiceGrpc;
+import com.google.common.flogger.StackSize;
+import com.google.tagpost.spanner.DataService;
+import io.grpc.Status;
 import io.grpc.stub.StreamObserver;
+import io.grpc.StatusRuntimeException;
 
+import java.util.List;
+import com.google.inject.Inject;
+import com.google.common.flogger.FluentLogger;
+
+/** Encapsulate all RPC methods of {@link TagpostServer} */
 public final class TagpostService extends TagpostServiceGrpc.TagpostServiceImplBase {
 
-    @Override
-    public void fetchMessage(FetchMessageRequest req, StreamObserver<FetchMessageResponse> responseObserver) {
-        FetchMessageResponse response = fetchMessageImpl(req);
-        responseObserver.onNext(response);
-        responseObserver.onCompleted();
+  private static final FluentLogger logger = FluentLogger.forEnclosingClass();
+  private DataService dataService;
+
+  @Inject
+  public TagpostService(DataService dataService) {
+    this.dataService = dataService;
+  }
+
+  @Override
+  public void fetchMessage(
+      FetchMessageRequest req, StreamObserver<FetchMessageResponse> responseObserver) {
+    FetchMessageResponse response = fetchMessageImpl(req);
+    responseObserver.onNext(response);
+    responseObserver.onCompleted();
+  }
+
+  @Override
+  public void fetchThreadsByTag(
+      FetchThreadsByTagRequest req, StreamObserver<FetchThreadsByTagResponse> responseObserver) {
+    try {
+      FetchThreadsByTagResponse response = fetchThreadsByTagImpl(req);
+      responseObserver.onNext(response);
+    } catch (Exception e) {
+      Status status = Status.UNKNOWN.withDescription(e.getMessage());
+      logger.atWarning().withCause(e).log("Fetch Threads By Tag Failed");
+      responseObserver.onError(status.asRuntimeException());
     }
+    responseObserver.onCompleted();
```

Comment:
Move this inside the `try` block, or make the `catch` block return early.

Reply:
Fixed.